gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapperTest.java

```diff
@@ -0,0 +1,224 @@
+package uk.nhs.adaptors.pss.translator.mapper;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.Condition;
+import org.hl7.fhir.dstu3.model.DateTimeType;
+import org.hl7.fhir.dstu3.model.Immunization;
+import org.hl7.fhir.dstu3.model.Observation;
+import org.hl7.fhir.dstu3.model.ReferralRequest;
+import org.hl7.v3.RCMRMT030101UK04Component4;
+import org.hl7.v3.RCMRMT030101UK04EhrComposition;
+import org.hl7.v3.RCMRMT030101UK04LinkSet;
+import org.hl7.v3.RCMRMT030101UK04ObservationStatement;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+import static org.springframework.util.ResourceUtils.getFile;
+import static uk.nhs.adaptors.pss.translator.testutil.XmlUnmarshallUtil.unmarshallFile;
+
+import lombok.SneakyThrows;
+
+@ExtendWith(MockitoExtension.class)
+public class ConditionMapperTest {
+    private static final String LINKSET_RESOURCES_BASE = "xml/Linkset/";
+    private static final String PATIENT_ID = "9014e659-f87e-43a2-b8c3-36fb91cc4225";
+    private static final String ENCOUNTER_ID = "77efedb3-e44e-43ff-9f81-23431a36df1b";
+    private static final String ASSERTER_ID = "9b8d9627-3e92-44c2-84eb-f37e4906c47e";
+    private static final String OBSERVATION_ID = "9f25604b-d2e8-402a-aae2-c78879c3e647";
+    private static final String LINKSET_ID = "DCC26FC9-4D1C-11E3-A2DD-010000000161";
+    private static final String CODING_DISPLAY = "THIS IS A TEST";
+    private static final String PRACTISE_CODE = "TEST123";
+    private static final Date EHR_EXTRACT_AVAILABILITY = new Date(2022, Calendar.JANUARY, 1);
+    private static final DateTimeType EHR_EXTRACT_AVAILABILITY_DATETIME = new DateTimeType(EHR_EXTRACT_AVAILABILITY);
+
+    private static final String ACTUAL_PROBLEM_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-ActualProblem-1";
+    private static final String PROBLEM_SIGNIFICANCE_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-ProblemSignificance-1";
+    private static final String RELATED_CLINICAL_CONTENT_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-RelatedClinicalContent-1";
+
+    @Mock
+    private CodeableConceptMapper codeableConceptMapper;
+    @Mock
+    private DateTimeMapper dateTimeMapper;
+
+    private ConditionMapper conditionMapper;
+
+//    List<Resource> relatedClinicalContent,
+//    Optional<String> encounterId,
```

Comment:
If these fields are required in the future, maybe add a TODO comment? If not, then they could be removed.

Reply:
removed these